includes/data-stores/class-wc-admin-reports-orders-data-store.php

```diff
@@ -42,15 +43,16 @@ class WC_Admin_Reports_Orders_Data_Store extends WC_Admin_Reports_Data_Store imp
 	 * @var array
 	 */
 	protected $report_columns = array(
-		'order_id'       => 'order_id',
-		'date_created'   => 'date_created',
-		'status'         => 'REPLACE(status, "wc-", "") as status',
-		'customer_id'    => 'customer_id',
-		'net_total'      => 'net_total',
-		'gross_total'    => 'gross_total',
-		'refund_total'   => 'refund_total',
-		'num_items_sold' => 'num_items_sold',
-		'customer_type'  => '(CASE WHEN returning_customer <> 0 THEN "returning" ELSE "new" END) as customer_type',
+		'order_id'         => 'order_id',
+		'date_created'     => 'date_created',
+		'date_created_gmt' => 'date_created_gmt',
```

Comment:
I know this is the way this is handled in core right now, but I thought there was a case recently where we considered using timestamps.  I might be misremembering though.  /cc @peterfabian

Are there problems we know of associated with storing in a timestamp and converting to local times after fetch? Or is it  less complicated just to match what's done in core?

Reply:
The problem was that the data is sliced into intervals the database (e.g. to days/weeks) and the boundaries of those intervals would be incorrect if time in the database was GMT/UTC. Having both local time and GMT time is possible, but then the local time needs to be updated when time zone changes and one needs to be careful how that works (e.g. I guess you can't just detect current GMT offset and update based on that, as you have DST etc).

Reply:
discussion with a bit of background: https://a8c.slack.com/archives/C0E1AV8T0/p1549463094980600

Reply:
Is there any reason against storing the GMT also here? Seems like it remedies the issue - I suppose it can create some edge cases if the stores tz changes as @joshuatf pointed out.

Reply:
This confused me too originally. TBH, changing store timezone could be a rarity we don't need to account for (just document). We could implement tracks to prove it's uncommon..

I'd suggest sticking to local timestamps for the stats tables, then flushing them on change, rather than trying to handle conversions of existing data.

Reply:
In case we don't want to store the GMT datetime in the stats table, I guess we could use the local datetime which is already stored and calculate the GMT time in the frontend given the store timezone offset. That will require some changes in the frontend but would avoid having to add a new column in the table. Do you think that would be a better solution @mikejolley?

Reply:
I think it's fine to add this to the table for efficiency as long as we understand the limitations and flush rather than try to convert should the timezone change :)

Reply:
> and flush rather than try to convert should the timezone change

The thing here is that when the user changes the store's timezone, we don't update any order in the database. It's only when the user edits one of the orders, that the datetimes of that order are updated according to the new timezone, that happens in `posts` and it's immediately replicated in `wc_order_stats`.

Considering that, what would be the point in flushing the `wc_order_stats` table when changing the timezone? We want datetimes from `posts` and `wc_order_stats` to be in sync and that's already happening now.

Am I missing something?

Reply:
I see what you mean. In that case I guess we do nothing; if posts and orders don't update either we just assume the dates in the DB are correct after a timezone change, even if in reality they are not..?

Reply:
@mikejolley that's how WooCommerce currently works, so I guess it makes sense wc-admin to do the same.